questions/shop.md

```diff
@@ -0,0 +1,104 @@
+# How to make a shop
+This is a quick guide of how to make a shop where you can buy items with other items or costing a scoreboard value (money or coins).
+
+## Item shop
+This method consists of buying items with other items, in this example, you will buy 1 netherite ingot with 5 diamonds.
+
+> [!NOTE]
+> You can add any command you wan't to run (for example a playsound) before the last command, but the selector must be the same as the command before, (`@a[tag=buy_netherite]` in this case)
+
+### Java
+There are 2 methods, one for pre-1.20.5 and the other for 1.20.5 and above, this is due to the new `execute if items` subcommand.
+> [!NOTE]
+> The method for pre-1.20.5 will still work in 1.20.5 and above.
+
+Both methods will require this setup:
+
+    # In chat / Load function
+    scoreboard objectives add diamonds dummy
+
+Both methods works the same:
+First we are going to know how many diamonds the player has storing the value in a scoreboard and if the player has that that number of diamonds we will add a tag (first and second command) or we will run a function instead if using a datapack.
+After that we are going to give the netherite ingot to the player with the tag and we will clear the diamonds from it.
+To avoid the player from a loop we will remove the tag as we dont need it more.
+
+#### Pre-1.20.5
+
+    # Command blocks
+    execute as @p store result score @s diamonds run clear @s diamond
+    execute as @p at @s[scores={diamonds=5..}] run tag @s add @s buy_netherite
+    give @a[tag=buy_netherite]  netherite_ingot 1
+    clear @a[tag=buy_netherite] diamond 5
+    tag @a remove buy_netherite
```

Comment:
This method should be changed completely.
Each time you select the closest player, the player selection starts from different positions, so it is possible that two command blocks select different players. So you need to add something like `positioned <button_pos>` before this so that one player is always selected. Or the first command is to select a buyer as
```
tag @p add buyer
```
It would also be nice if you added a message for each method to send a message to the chat / actionbar if the player does not have enough diamonds.

Reply:
You give the `buyer_netherite` tag, but don't remove it. Also, you don't need to add more tags, just use the score check. There is also no need to complicate the JSON text.
```
tag @p add buyer.netherite
execute as @a[tag=buyer.netherite] store result score @s diamonds run clear @s diamond 0
give @a[tag=buyer.netherite,scores={diamonds=5..}] netherite_ingot 1
clear @a[tag=buyer.netherite,scores={diamonds=5..}] diamond 5
tellraw @a[tag=buyer.netherite,scores={diamonds=5..}] {"text":"You bought a netherite ingot for 5 diamonds","color":"green"}
tellraw @a[tag=buyer.netherite,scores={diamonds=..4}] {"text":"You don't have 5 diamonds","color":"dark_red"}
tag @a remove buyer.netherite
```